test/AutoRest.CSharp.UnitTests/Common/Generation/Types/CSharpTypeTests.cs

```diff
@@ -175,5 +176,42 @@ public void GetGenericTypeDefinitionForConstructedType()
             var actual = new CSharpType(typeof(List<>), typeof(string)).GetGenericTypeDefinition();
             Assert.AreEqual(new CSharpType(typeof(List<>)), actual);
         }
+        [TestCase(typeof(int), false)]
+        [TestCase(typeof(AsyncPageable<>), true)]
+        [TestCase(typeof(AsyncPageable<int>), true)]
+        [TestCase(typeof(TestAsyncPageable<>), false)]
+        [TestCase(typeof(TestAsyncPageable<int>), false)]
+        [TestCase(typeof(Pageable<>), false)]
+        [TestCase(typeof(Pageable<int>), false)]
+        [TestCase(typeof(TestPageable<>), false)]
+        [TestCase(typeof(TestPageable<int>), false)]
+        public void IsAsyncPageable(Type type, bool expected)
+            => Assert.AreEqual(new CSharpType(type).IsAsyncPageable, expected);
+
+        [TestCase(typeof(int), false)]
+        [TestCase(typeof(AsyncPageable<>), false)]
+        [TestCase(typeof(AsyncPageable<int>), false)]
+        [TestCase(typeof(TestAsyncPageable<>), false)]
+        [TestCase(typeof(TestAsyncPageable<int>), false)]
+        [TestCase(typeof(Pageable<>), true)]
+        [TestCase(typeof(Pageable<int>), true)]
+        [TestCase(typeof(TestPageable<>), false)]
```

Comment:
Why TestPageable<> is not 'Pageable' here?

Reply:
This represents a derived type for `Pageable<T>`, and we need this method to return false.

This is not new test case, I just move the test from another class here